tests/http_spec.rs

```diff
@@ -0,0 +1,216 @@
+use std::collections::BTreeMap;
+use std::fs;
+use std::path::Path;
+use std::str::FromStr;
+use std::sync::Arc;
+
+use anyhow::anyhow;
+use async_graphql_value::ConstValue;
+use hyper::{Body, Request};
+use reqwest::header::{HeaderName, HeaderValue};
+use serde::Deserialize;
+use serde_json::Value;
+use tailcall::blueprint::Blueprint;
+use tailcall::config::Config;
+use tailcall::http::{graphql_request, HttpClient, Method, Response, ServerContext};
+use url::Url;
+
+#[derive(Deserialize, Clone, Debug)]
+pub enum Annotation {
+  Skip,
+  Only,
+  Fail,
+}
+#[derive(Deserialize, Clone, Debug, Eq, PartialEq)]
+pub struct APIRequest {
+  #[serde(default)]
+  method: Method,
+  pub url: Url,
+  #[serde(default)]
+  pub headers: BTreeMap<String, String>,
+  #[serde(default)]
+  pub body: serde_json::Value,
+}
+#[derive(Deserialize, Clone, Debug)]
+pub struct APIResponse {
+  pub status: u16,
+  #[serde(default)]
+  pub headers: BTreeMap<String, String>,
+  #[serde(default)]
+  pub body: serde_json::Value,
+}
+#[derive(Deserialize, Clone, Debug, Eq, PartialEq)]
+pub struct UpstreamRequest(pub APIRequest);
+#[derive(Deserialize, Clone, Debug)]
+pub struct UpstreamResponse(APIResponse);
+#[derive(Deserialize, Clone)]
+pub struct DownstreamRequest(pub APIRequest);
+#[derive(Deserialize, Clone)]
+pub struct DownstreamResponse(pub APIResponse);
+#[derive(Deserialize, Clone)]
+pub struct DownstreamAssertion {
+  pub request: DownstreamRequest,
+  pub response: DownstreamResponse,
+  pub annotation: Option<Annotation>,
```

Comment:
I think the annotation should be at Spec level. The spec succeeds or fails. It's possible that we apply a `skip` annotation for an assertion, but because the mock working correctly because of a batching bug, the spec throws up an error.

Reply:
done